rules-reviewed/eap6/uncategorized/environment-dependent.windup.xml

```diff
@@ -23,7 +23,7 @@
                 </javaclass>
             </when>
             <perform>
-                 <hint title="Dynamic class instantiation" effort="0" category-id="optional">
+                 <hint title="Dynamic class instantiation" effort="1" category-id="optional">
```

Comment:
These are reasonable points, but:

-- We report a top-line story points number (in large font on App List and App Details), and this count includes effort for all issues, not just mandatory ones. Effort counts from different categories of issues are also aggregated in other places, like the per-jar sections on App Details. Maybe the reports should not do this, and should highlight effort counts that only take into account for mandatory issues? But for now that's how things are counted.

-- Story point estimates dominated by many appearances of one issue type are confusing, as they imply that >90% of the migration effort comes from that one type of issue, and drown out the other types of issues present in the app.

We really should make some changes to the way effort is counted and reported, and this should probably be a focus in the next release cycle after the initial release of windup-web. Until then, though, we really can't merge this change. It will change story point estimates to something that is very different from previous versions of Windup and confusing to most users.

My suggestion for rules like this, where you've proposed a change but we couldn't merge it, is to use the rules override feature that was introduced in Windup 2.5. This will let you maintain a personal version of the rules that overrides some of our decisions not to incorporate some changes into the core rulesets and keeps your changes intact as you pull in updates to the basic rules.